src/Sarif.WorkItems/SarifWorkItemsExtensions.cs

```diff
@@ -103,20 +104,47 @@ private static string ConstructFullRuleIdentifier(ReportingDescriptor reportingD
             return fullRuleIdentifier;
         }
 
-        public static string CreateWorkItemDescription(this SarifLog log)
+        public static string CreateWorkItemDescription(this SarifLog log, Uri locationUri)
         {
-            Dictionary<string, int> resultCountsByTool = ComputeToolResultCounts(log);
-            StringBuilder templateText = new StringBuilder(@"This bug contains results for the below tool(s) and issue(s).  It was filed automatically.");
-            templateText.AppendLine();
-            templateText.AppendLine();
+            Dictionary<Run, int> resultCountsByRun = ComputeRunResultCounts(log);
+            StringBuilder templateText = new StringBuilder();
+            int runningResults = 0;
+            string toolNames = string.Empty;
+            string multipleToolsFooter = string.Empty;
+
+            Uri runRepositoryUri = resultCountsByRun.FirstOrDefault().Key?.VersionControlProvenance?.FirstOrDefault().RepositoryUri;
+            string detectionLocation = !string.IsNullOrEmpty(runRepositoryUri?.OriginalString) ? runRepositoryUri?.OriginalString : locationUri?.OriginalString;
```

Comment:
>detectionLocation [](start = 19, length = 17)

BUG? What has the repo URI (_e.g_, `[https://github.com/microsoft/sarif-sdk`](https://github.com/microsoft/sarif-sdk`)) to do with the detection location(_e.g_, `src\Sarif\Cache.cs`)?

Reply:
That's directly from Mike's template, it's the general pointer to what was being scanned.  If it's null, then we use the filename from the location.  Since the scan is likely to cover multiple issues over multiple files, I agree that this makes a good general description assuming it's present.

---
In reply to: [398214479](https://github.com/microsoft/sarif-sdk/pull/1823#discussion_r398214479) [](ancestors = 398214479)

Reply:
Thanks for the explanation. I now understand the rationale for including the repo URI. But as the code stands, if repo URI is present, you'll use it _even if there's only one location_. You can't tell at this point in the code if there's more than one location, and you need to know to decide whether to mention the repo URL.

---
In reply to: [398216755](https://github.com/microsoft/sarif-sdk/pull/1823#discussion_r398216755) [](ancestors = 398216755,398214479)

Reply:
If there's a bug here, it's very much as specc'ed.  The design is to default to the repo URL, and only mention other locations if that is empty, even in the single location case.  I believe the intent of the design is that the Sarif attachement will contain the specifics needed for the location.

---
In reply to: [398223529](https://github.com/microsoft/sarif-sdk/pull/1823#discussion_r398223529) [](ancestors = 398223529,398216755,398214479)

Reply:
First, of course, the specification may have rough edges to thinking. :) The rationale here is that for a single location issue, the title already contains the issue. And so it is more valuable in the description to add information about what repo/branch/commit was scanned. Note that this information isn't surfaced in any of the viewers, you have to look at the raw SARIF to get it. So worth raising visibility on it if available. If it isn't, we recap the first location (and specifically call out if there are other locations to examine, without listing them in detail).

---
In reply to: [398271699](https://github.com/microsoft/sarif-sdk/pull/1823#discussion_r398271699) [](ancestors = 398271699,398223529,398216755,398214479)

Reply:
Marking "Won't Fix" since we will surely give more thought to this question, especially as customers comment on the quality of the bugs we file.

---
In reply to: [398679769](https://github.com/microsoft/sarif-sdk/pull/1823#discussion_r398679769) [](ancestors = 398679769,398271699,398223529,398216755,398214479)